src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Security/Services/RoleServiceExtensions.cs

```diff
@@ -13,14 +14,25 @@ public static async Task<IEnumerable<string>> GetRoleNamesAsync(this IRoleServic
         return roles.Select(r => r.RoleName);
     }
 
+    public static async Task<IEnumerable<string>> GetRoleNamesAsync(this IRoleService roleService, Func<IRole, bool> predicate)
```

Comment:
I would advise against starting to wrap linq extensions in other extensions. Never ending story otherwise, and not much gain. Removing a "where" in the caller in not worth it IMO, what it made me do is only wonder what it was doing internally (a query or a filter), and then have to drill into the implementation. And knowing it was using linq would let developers react to its implications.

Reply:
agreed. I guess having the existing GetRoleNameAsync() and other expection dragged me into doing this without thinking about it too much. we should at some point obsolete the existing extensions.